File: 455.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        # time: O(mlogm + nlogn), space: O(m + n)
        g.sort(), s.sort()
        gIdx, sIdx = 0, 0
        contentChildren = 0
        while gIdx < len(g) and sIdx < len(s):
            if g[gIdx] <= s[sIdx]:
                contentChildren += 1
                gIdx += 1
            sIdx += 1
        return contentChildren

File: 455.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        // time: O(mlogm + nlogn), space: O(m + n)
        Arrays.sort(g);
        Arrays.sort(s);
        int contentChildren = 0;
        int idx = 0;
        while (contentChildren < g.length && idx < s.length) {
            if (g[contentChildren] <= s[idx]) {
                contentChildren++;
            }
            idx++;
        }
        return contentChildren;
    }
}